tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,154 @@
+		$this->assertEquals( 'running', constant( $job_class . '::JOB_STATUS_RUNNING' ) );
+		$this->assertEquals( 'partial', constant( $job_class . '::JOB_STATUS_PARTIAL' ) );
+		$this->assertEquals( 'completed', constant( $job_class . '::JOB_STATUS_COMPLETE' ) );
+		$this->assertEquals( 'failed', constant( $job_class . '::JOB_STATUS_FAILED' ) );
+	}
+
+	/**
+	 * @covers ::set_status
+	 */
+	public function test_set_status_false_for_invalid_status() {
+		$result = $this->job->set_status( 'invalid_status' );
+		$this->assertFalse( $result );
+	}
+
+	public function test_create_job() {
+		$this->assertInstanceOf( Perflab_Background_Job::class, $this->job );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_run_for_job() {
+		$run = $this->job->should_run();
+		$this->assertTrue( $run );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_not_run_for_completed_job() {
+		// Mark job as complete.
+		$this->job->set_status( 'completed' );
+		$run = $this->job->should_run();
+
+		$this->assertFalse( $run );
+	}
+
+	/**
+	 * @covers ::lock
+	 * @covers ::unlock
+	 */
+	public function test_lock_unlock() {
+		$time = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+		$this->job->unlock();
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertEmpty( $lock_time );
+	}
+
+	/**
+	 * @covers ::get_start_time
+	 */
+	public function test_get_start_time() {
+		$time = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+
+		$start_time = $this->job->get_start_time();
+		$this->assertEquals( $start_time, $lock_time );
+		$this->assertEquals( $start_time, $time );
+		$this->assertEquals( $time, $lock_time );
+	}
+
+	public function test_max_attempts_filter() {
+		$filter_ran = false;
+		add_filter(
+			'perflab_job_max_attempts_allowed',
+			function( $value ) use ( &$filter_ran ) {
+				$filter_ran = true;
+				return $value;
+			}
+		);
+		$this->job->should_run();
+
+		$this->assertTrue( $filter_ran );
+	}
+
+	public function test_max_attempts_limit_reached() {
+		$before_attempts_limit = $this->job->should_run();
+		update_term_meta( $this->job->get_id(), $this->job::META_KEY_JOB_ATTEMPTS, 2 );
+		add_filter(
+			'perflab_job_max_attempts_allowed',
+			function() {
+				return 1;
+			}
+		);
+		$after_attempts_limit = $this->job->should_run();
+
+		$this->assertTrue( $before_attempts_limit );
+		$this->assertFalse( $after_attempts_limit );
+	}
+
+	/**
+	 * @covers ::set_status
+	 * @dataProvider job_provider
+	 */
+	public function test_set_status_false_for_valid_status( $status ) {
+		$this->assertTrue( $status );
+		$this->assertTrue( $status );
+		$this->assertTrue( $status );
+		$this->assertTrue( $status );
```

Comment:
This now asserts the same thing 4 times, we only need it once :)

Reply:
Oops! I missed to remove other 3. 
Fixed it, thanks :)